lib/rules/declaration-property-value-keyword-no-deprecated/index.mjs

```diff
@@ -0,0 +1,174 @@
+import valueParser from 'postcss-value-parser';
+
+import { isPlainObject, isRegExp, isString } from '../../utils/validateTypes.mjs';
+import { declarationValueIndex } from '../../utils/nodeFieldIndices.mjs';
+import getDeclarationValue from '../../utils/getDeclarationValue.mjs';
+import isStandardSyntaxDeclaration from '../../utils/isStandardSyntaxDeclaration.mjs';
+import optionsMatches from '../../utils/optionsMatches.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import setDeclarationValue from '../../utils/setDeclarationValue.mjs';
+import { singleValueColorProperties } from '../../reference/properties.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'declaration-property-value-keyword-no-deprecated';
+
+const messages = ruleMessages(ruleName, {
+	expected: (actual, expected) => `Expected "${actual}" to be "${expected}"`,
+	rejected: (property, keyword) =>
+		`Unexpected deprecated keyword "${keyword}" for property "${property}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/declaration-property-value-keyword-no-deprecated',
+	fixable: true,
+};
+
+const DEPRECATED_COLORS = {
+	activecaption: 'canvas',
+	appworkspace: 'canvas',
+	background: 'canvas',
+	inactivecaption: 'canvas',
+	infobackground: 'canvas',
+	menu: 'canvas',
+	scrollbar: 'canvas',
+	window: 'canvas',
+	activeborder: 'ButtonBorder',
+	inactiveborder: 'ButtonBorder',
+	threeddarkshadow: 'ButtonBorder',
+	threedhighlight: 'ButtonBorder',
+	threedlightshadow: 'ButtonBorder',
+	threedshadow: 'ButtonBorder',
+	windowframe: 'ButtonBorder',
+	captiontext: 'CanvasText',
+	infotext: 'CanvasText',
+	menutext: 'CanvasText',
+	windowtext: 'CanvasText',
+	buttonhighlight: 'ButtonFace',
+	buttonshadow: 'ButtonFace',
+	threedface: 'ButtonFace',
+	inactivecaptiontext: 'GrayText',
+};
+const COLOR_PROPERTIES = [...singleValueColorProperties.values()].map(
+	(value) => /** @type {[string, Record<string, string>]} */ ([value, DEPRECATED_COLORS]),
+);
+const PROPERTY_NAME_TO_KEYWORDS = new Map([
+	[
+		'appearance',
+		/** @type {string[] | Record<string, string>} */ ({
```

Comment:
[suggestion] To simplify the logic, how about separating the constant into two objects like this?

`PROPERTY_NAME_TO_KEYWORDS` => `PROPERTY_NAME_TO_KEYWORD_MAPS` and `PROPERTY_NAME_TO_KEYWORD_LISTS`

```suggestion
const PROPERTY_NAME_TO_KEYWORD_MAPS = new Map([
	[
		'appearance',
		{
```

Reply:
I prefer having only one map because:

- we have only one getter
- it will be simpler to reason about if we chose to support shorthands later on (2 paths instead or 3)
- I find it clearer to read them all at a glance